bootstrap/bootstrap.go

```diff
@@ -319,132 +335,296 @@ func (b *Bootstrap) StartAPIServer(ctx context.Context) error {
 	return nil
 }
 
-func (b *Bootstrap) StopProfilerServer() {
-	if b.profiler == nil {
-		return
+func (fnb *EVMGatewayNodeBuilder) initStorage() error {
+	logger := fnb.Logger
+	cfg := fnb.Config
+
+	store := pebble2.New(fnb.DB, fnb.Logger)
+
+	blocks := pebble2.NewBlocks(store, cfg.FlowNetworkID)
+	storageAddress := evm.StorageAccountAddress(cfg.FlowNetworkID)
+	registerStore := pebble2.NewRegisterStorage(store, storageAddress)
+
+	// hard set the start cadence height, this is used when force reindexing
+	if cfg.ForceStartCadenceHeight != 0 {
+		logger.Warn().Uint64("height", cfg.ForceStartCadenceHeight).Msg("force setting starting Cadence height!!!")
+		if err := blocks.SetLatestCadenceHeight(cfg.ForceStartCadenceHeight, nil); err != nil {
+			return err
+		}
 	}
 
-	b.logger.Warn().Msg("shutting down profiler server")
+	// if database is not initialized require init height
+	if _, err := blocks.LatestCadenceHeight(); errors.Is(err, errs.ErrStorageNotInitialized) {
+		// TODO(JanezP): move this to a separate function
+		err = func() (innerErr error) {
+			batch := store.NewBatch()
+			defer func(batch *pebble.Batch) {
+				innerErr = batch.Close()
+			}(batch)
+
+			cadenceHeight := cfg.InitCadenceHeight
+			evmBlokcHeight := uint64(0)
+			cadenceBlock, err := fnb.Client.GetBlockHeaderByHeight(context.Background(), cadenceHeight)
+			if err != nil {
+				return fmt.Errorf("could not fetch provided cadence height, make sure it's correct: %w", err)
+			}
 
-	err := b.profiler.Stop()
-	if err != nil {
-		if errors.Is(err, context.DeadlineExceeded) {
-			b.logger.Warn().Msg("Profiler server graceful shutdown timed out")
-			b.profiler.Close()
-		} else {
-			b.logger.Err(err).Msg("Profiler server graceful shutdown failed")
+			snapshot, err := registerStore.GetSnapshotAt(evmBlokcHeight)
+			if err != nil {
+				return fmt.Errorf("could not get register snapshot at block height %d: %w", 0, err)
+			}
+
+			delta := storage.NewRegisterDelta(snapshot)
+			accountStatus := environment.NewAccountStatus()
+			err = delta.SetValue(
+				storageAddress[:],
+				[]byte(flowGo.AccountStatusKey),
+				accountStatus.ToBytes(),
+			)
+			if err != nil {
+				return fmt.Errorf("could not set account status: %w", err)
+			}
+
+			err = registerStore.Store(delta.GetUpdates(), evmBlokcHeight, batch)
+			if err != nil {
+				return fmt.Errorf("could not store register updates: %w", err)
+			}
+
+			if err := blocks.InitHeights(cadenceHeight, cadenceBlock.ID, batch); err != nil {
+				return fmt.Errorf(
+					"failed to init the database for block height: %d and ID: %s, with : %w",
+					cadenceHeight,
+					cadenceBlock.ID,
+					err,
+				)
+			}
+
+			err = batch.Commit(pebble.Sync)
+			if err != nil {
+				return fmt.Errorf("could not commit register updates: %w", err)
+			}
+
+			logger.Info().
+				Stringer("fvm_address_for_evm_storage_account", storageAddress).
+				Msgf("database initialized with cadence height: %d", cadenceHeight)
```

Comment:
maybe also log the evm Height

Reply:
This is the bootstrapping code branch, so evm height will be 0